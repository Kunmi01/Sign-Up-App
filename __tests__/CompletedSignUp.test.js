import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import ConnectedCompletedSignUp, {
  CompletedSignUp
} from '../src/components/SignUp/components/CompletedSignUp';

describe('CompletedSignUp', () => {
  const initialState = {
    userData: {
      name: '',
      role: '',
      email: '',
      password: ''
    },
    privacyData: {
      trayUpdates: null,
      productEmails: null
    }
  };
  const mockStore = configureStore();
  let store;
  let container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ConnectedCompletedSignUp />
        </BrowserRouter>
      </Provider>
    );
  });

  it('renders connected(SMART) component', () => {
    expect(container.find(ConnectedCompletedSignUp).exists()).toBe(true);
    expect(toJson(container)).toMatchSnapshot();
  });

  it('prop match with initial store state', () => {
    const completedSignUpContainer = container.find(CompletedSignUp);
    expect(completedSignUpContainer.prop('state')).toEqual(initialState);
  });
});
