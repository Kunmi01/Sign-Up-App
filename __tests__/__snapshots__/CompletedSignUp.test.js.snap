// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompletedSignUp renders connected(SMART) component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(CompletedSignUp)>
        <CompletedSignUp
          dispatch={[Function]}
          state={
            Object {
              "privacyData": Object {
                "productEmails": null,
                "trayUpdates": null,
              },
              "userData": Object {
                "email": "",
                "name": "",
                "password": "",
                "role": "",
              },
            }
          }
        >
          <div
            className="completed-sign-up"
          >
            <div
              className="completed-sign-up__wrapper"
            >
              <h3
                className="completed-sign-up__header"
              >
                Completed
              </h3>
              <p
                className="completed-sign-up__message"
              >
                Please verify your email address, you should have received an email from us already
              </p>
            </div>
          </div>
        </CompletedSignUp>
      </Connect(CompletedSignUp)>
    </Router>
  </BrowserRouter>
</Provider>
`;
